src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,26 +105,48 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
+  }, [numberOfImgs, randomPhotos]);
+
+  type unsplashPhotoFortmat = {
+    id: string;
+    url: string;
+    alt: string;
+    urls: { regular: string };
+    alt_description: string;
+  };
+
+  /*----------------------FETCH RANDOM IMAGES ---------------------- */
 
-  // to avoid bug undefined url when user change number of images
   useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
+    async function getData() {
```

Comment:
This can also be an arrow function. Generally, in React we tend to use arrow functions over function declarations. It's more concise and useful.